### ISSUE TITLE
Campus results

### ISSUE THREAD

**Issue by rock19970106** [rayguan97/GANav-offroad#4]
Hello, I test my campus dataset captured by an Intel realsense D435 (image size : 640*480) with your pre-trained rugd_group6 model. The result is not so good. Here are some results: (only show the smooth region)

![003567](https://user-images.githubusercontent.com/49554477/145668698-53581aab-75e1-4ad0-b381-08e9c2248cf8.png)
![005608](https://user-images.githubusercontent.com/49554477/145668702-50420e0b-c216-40cd-891f-cf1dfcbcfaf6.png)
![006053](https://user-images.githubusercontent.com/49554477/145668710-a6b09d2e-d260-4de1-9a72-976bcd66f09f.png)
![006487](https://user-images.githubusercontent.com/49554477/145668716-5c514e7a-386b-4149-832a-cf5dc2133a45.png)
![006807](https://user-images.githubusercontent.com/49554477/145668723-c12b5024-763a-403b-afe5-e0523da6d383.png)
![008548](https://user-images.githubusercontent.com/49554477/145668731-5a5426eb-72a3-441c-ad04-394fa173330b.png)
![014043](https://user-images.githubusercontent.com/49554477/145668736-9bd25429-4ba7-4a2d-9294-ddc0bd2e29ef.png)
![014634](https://user-images.githubusercontent.com/49554477/145668740-45ac407e-2f04-4af8-a3f3-3b5f41295c82.png)

I'm wondering if the "Domain Gap" can make such big difference.

**Comment by rock19970106** [comment]
Do you have fine-tuned model for your campus test?

**Comment by rayguan97** [comment]
Please refer to Fig 3 of the paper, which is the results of GANav on our campus. We only shared a few examples but it worked very well during our testing in general. Here is some [footages](https://youtu.be/wy4k7Oz1HHM) and we even got some better ones from recent experiments. It's not perfect since there is domain gap, but it should not be that worse..

There are Domain gap between two different datasets, so it's possible that it did not work well on your dataset. A few possible reasons are:
1. lighting condition
2. angle of the camera: how tall is the camera and what is the angle of the camera. Is the image capture on the robot, or captured in hand.
3. Structure of the picture: our camera is facing towards the ground and only have very small portion of the background.

One thing I noticed is that your prediction result is missing many regions, but what we have provided did not work like this. What kind of modifications have you made? Or could there be something wrong during deployment? I believe that it should work well since the visual features are rather distinct and we are actually training on asphalt terrain as well. This is very strange.

We did not fine-tuned our model as we do not have any annotation on our campus images. If you have some annotation tools and resources, I recommend that you could annotate some images from your campus and fine-tune it on GANav and see if the performance can improve.

**Comment by rock19970106** [comment]
Hi, thank you very much. The predicted regions are not missing, I only visualized the smooth region. I think the main reason may be the mounting height of our camera, about 60~70cm or the angle of the camera , the pictures contains too much background.

**Comment by rock19970106** [comment]
The mounting height of your camera seems about only 20-30cm

**Comment by rock19970106** [comment]
Hi， here are 10 sample images, can you test it for me. I want to know if there is someting wrong during deployment.
[campus.zip](https://github.com/rayguan97/GANav-offroad/files/7697078/campus.zip)

**Comment by rayguan97** [comment]
Hey, it appears that your GANav deployment is fine -- from you image, I also got pretty bad segmentation results. :( However, I got one that looks decent at least. Here:

![000007](https://user-images.githubusercontent.com/19572733/145759792-41f8e59e-da4e-4c72-a708-4df331a81e01.png)
![000007_pred](https://user-images.githubusercontent.com/19572733/145759824-f65a705d-4d9f-4c97-b1dc-ed71196582fd.png)

But the following aspects could also be important factors, as mentioned here:
"

1.lighting condition
2.angle of the camera: how tall is the camera and what is the angle of the camera. Is the image capture on the robot, or captured in hand.
3. Structure of the picture: our camera is facing towards the ground and only have very small portion of the background."

To sum up, your GANav setting is fine, but maybe the hardware deployment could be a issue.

**Comment by rock19970106** [comment]
okay, thank you very much!

**Comment by rock19970106** [comment]
Hi,  dou you incline your camera to face towards the ground? if so, what's the angle of inclination?

**Comment by rock19970106** [comment]
Hi, dou you incline your camera to face towards the ground? if so, what's the angle of inclination?

I readjust the mounting height of our camera, but the performance is also not so good, espically on the relatively empty asphalt road. On the contrary, it works well in some cement roads with strong structural features.

**Comment by rock19970106** [comment]
Hello, here are some test results. We mount the realsense D435 camera at a height of  about 35cm， and the camera is placed parallel to the ground. do you incline your camera to face towards the ground? if so, what's the angle of inclination?

[campus test.zip](https://github.com/rayguan97/GANav-offroad/files/7727438/campus.test.zip)

**Comment by rayguan97** [comment]
The height of our camera is 25 to 30 cm and the pitch angle is around 15-20 degrees towards the ground. I check those pictures but the portion of the background is too high.

If it still don't work well, maybe the domain gap is too big. In our case the real-world results are not perfect but still useful.

**Comment by rock19970106** [comment]
okay, thank you very much. I will readjust the pitch angle and try to test again.

**Comment by rock19970106** [comment]
Hello, I readjust the pitch angle and test. The results is good and useful. Thank you very much!

**Comment by rayguan97** [comment]
Thank you! Let me know if you have any more issues!
